spec/controllers/api/v1/canned_responses_controller_spec.rb

```diff
@@ -0,0 +1,125 @@
+require 'rails_helper'
+
+RSpec.describe 'Canned Responses API', type: :request do
+  let(:account) { create(:account) }
+
+  before do
+    create(:canned_response, account: account)
+  end
+
+  describe 'GET /api/v1/canned_responses' do
+    context 'when it is an unauthenticated user' do
+      it 'returns unauthorized' do
+        get '/api/v1/canned_responses'
+
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    context 'when it is an authenticated user' do
+      let(:agent) { create(:user, account: account, role: :agent) }
+
+      it 'returns all the canned responses' do
+        get '/api/v1/canned_responses',
+            headers: agent.create_new_auth_token,
+            as: :json
+
+        expect(response).to have_http_status(:success)
+        expect(JSON.parse(response.body)).to eq([CannedResponse.last].as_json)
```

Comment:
I guess `account.canned_responses` would be a better approach than `CannedResponse`

Reply:
Yes, that is true. Will change it :)